google/cloud/bigtable/data/_async/_mutate_rows.py

```diff
@@ -89,21 +90,21 @@ def __init__(
             bt_exceptions._MutateRowsIncomplete,
         )
         # build retryable operation
-        retry = retries.AsyncRetry(
+        retry_wrapped = functools.partial(
+            retries.retry_target,
```

Comment:
I have been looking at https://www.learnpython.org/en/Partial_functions to try to understand this a little bit better. I guess we are expecting this `retries.retry_target` to be in place of the `retry` function defined before?

Reply:
Yes, this will be a little complicated without context. 

AsyncRetry is just a [wrapper around retries.retry_target](https://github.com/googleapis/python-api-core/blob/6251eab3fca5f7e509cb9b6e476ce1184094b711/google/api_core/retry_async.py#L223). But AsynRetry doesn't expose the sleep_generator argument we need to customize here. So I'm replacing the AsyncRetry instance with the lower-level [retries.retry_target](https://github.com/googleapis/python-api-core/blob/6251eab3fca5f7e509cb9b6e476ce1184094b711/google/api_core/retry_async.py#L74).

The partial allows us to prepare the function's arguments without calling it yet. In the future, we can call `retry_wrapped` without any arguments, and it will call it with everything we passed into the partial here. This is useful for things like retries, where we may not always have all the context to re-build the entire operation when we want to launch a new attempt